community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/ShortestPathComplexQueryAcceptanceTest.scala

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.internal.cypher.acceptance
+
+import org.hamcrest.CoreMatchers._
+import org.junit.Assert._
+import org.neo4j.cypher.internal.RewindableExecutionResult
+import org.neo4j.cypher.internal.compatibility.ExecutionResultWrapperFor3_0
+import org.neo4j.cypher.{ExecutionEngineFunSuite, NewPlannerTestSupport}
+import org.neo4j.graphdb.Node
+
+class ShortestPathComplexQueryAcceptanceTest extends ExecutionEngineFunSuite with NewPlannerTestSupport {
+
+  test("allShortestPaths with complex LHS should be planned with exhaustive fallback and include predicate") {
+    setupModel()
+    val result = executeUsingCostPlannerOnly(
+      """
+        |//same idea but iterating over two collections of nodes
+        |profile match (charles:Pixie { fname : 'Charles'}),(joey:Pixie { fname : 'Joey'}),(kim:Pixie { fname : 'Kim'})
+        |with kim as kimDeal, collect(charles) as charlesT, collect(joey) as joeyS
+        |unwind charlesT AS charlesThompson
+        |unwind joeyS AS joeySantiago
+        |match pathx = allShortestPaths((charlesThompson)-[*1..5]-(joeySantiago))
+        |where none (n IN nodes(pathx) where id(n) = id(kimDeal))
+        |with nodes(pathx) as nodes
+        |return nodes
+      """.stripMargin)
+    val results = result.columnAs("nodes").toList
+    println(results)
+    println(result.executionPlanDescription())
+    val ids = results(0).asInstanceOf[Seq[_]].map(n => n.asInstanceOf[Node].getId)
```

Comment:
If we are asserting on the ids only, isn't it easier to adapt the query to return only ids, instead of this mouthful?

E.g:

```
...
RETURN [n IN nodes | id(n)] AS ids
```

and then

```
result.toList should equal List(Map("ids" -> List(0, 4, 3, 2)))
```

That also makes sure that there isn't a second row returned (which is overlooked by only asserting on `results(0)`).